### PR TITLE
migrate requestheader_controller.go to structured logs

### PR DESCRIPTION
**Pull request by iakisme**
migrate pkg/authentication/request/headerrequest/requestheader_controller.go to structured logs

**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

xref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by wangkai1994** [comment]
/assign deads2k
/assign wojtek-t

some fix, thanks

**Comment by wangkai1994** [comment]
@enj small cleanup, please have a quick look, Thanks

**Comment by yangjunmyfm192085** [comment]
I think you need to confirm the  format of migrate logs again.
Maybe the following description is helpful.
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go [hunk 1]
```diff
@@ -241,7 +241,7 @@ func (c *RequestHeaderAuthRequestController) syncConfigMap(configMap *corev1.Con
 	}
 	if hasChanged {
 		c.exportedRequestHeaderBundle.Store(newRequestHeaderBundle)
-		klog.V(2).Infof("Loaded a new request header values for %v", c.name)
+		klog.V(2).InfoS("Loaded a new request header values for", c.name)
```
Comment (enj) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:244]: Wrong format.

```suggestion
		klog.V(2).InfoS("Loaded new request header values", "name", c.name)
```

Comment (wangkai1994) [reply staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:244]: thanks for your help! you are so nice

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go [hunk 2]
```diff
@@ -241,7 +241,7 @@ func (c *RequestHeaderAuthRequestController) syncConfigMap(configMap *corev1.Con
 	}
 	if hasChanged {
 		c.exportedRequestHeaderBundle.Store(newRequestHeaderBundle)
-		klog.V(2).Infof("Loaded a new request header values for %v", c.name)
+		klog.V(2).InfoS("Loaded new request header values", "name", c.name)
```
Comment (yangjunmyfm192085) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:244]: Line193，please modify together,thanks.

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go [hunk 3]
```diff
@@ -166,8 +166,8 @@ func (c *RequestHeaderAuthRequestController) Run(workers int, stopCh <-chan stru
 	defer utilruntime.HandleCrash()
 	defer c.queue.ShutDown()
 
-	klog.Infof("Starting %s", c.name)
-	defer klog.Infof("Shutting down %s", c.name)
+	klog.InfoS("Starting", c.name)
```
Comment (shubheksha) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:169]:
```suggestion
	klog.InfoS("Starting", "controllerName", c.name)
```

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go [hunk 4]
```diff
@@ -166,8 +166,8 @@ func (c *RequestHeaderAuthRequestController) Run(workers int, stopCh <-chan stru
...
+	defer klog.InfoS("Shutting down", c.name)
```
Comment (shubheksha) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:170]:
```suggestion
	defer klog.InfoS("Shutting down", "controllerName", c.name)
```

File: staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go (hunk 2 above)
Comment (shubheksha) [inline staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader_controller.go:244]:
```suggestion
		klog.V(2).InfoS("Loaded new request header values", "controllerName", c.name)
```

**Review by shubheksha — CHANGES_REQUESTED** [review]

**Comment by wangkai1994** [comment]
@yangjunmyfm192085 @shubheksha thank you guys. i am so busy in my job.

**Pull request closed without merging**